test/fakeintake/aggregator/connectionsAggregator.go

```diff
@@ -0,0 +1,70 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package aggregator
+
+import (
+	"fmt"
+
+	agentmodel "github.com/DataDog/agent-payload/v5/process"
+
+	"github.com/DataDog/datadog-agent/test/fakeintake/api"
+)
+
+// Connections type contain all payload from /api/v1/connections
+type Connections struct {
+	agentmodel.CollectorConnections
+}
+
+// name return connection payload name based on hostname and network ID
+func (c *Connections) name() string {
+	return c.HostName + c.NetworkId
+}
+
+// GetTags return tags connection
+func (c *Connections) GetTags() []string {
+	dns, err := c.GetDNSNames()
+	if err != nil {
+		return []string{}
+	}
+	return dns
+}
+
+// DecodeCollectorConnection return a CollectorConnections protobuf object from raw bytes
+func DecodeCollectorConnection(b []byte) (cnx *agentmodel.CollectorConnections, err error) {
+	m, err := agentmodel.DecodeMessage(b)
+	if err != nil {
+		return nil, err
+	}
+	conns, ok := m.Body.(*agentmodel.CollectorConnections)
+	if !ok {
+		return nil, fmt.Errorf("not protobuf process.CollectorConnections type")
+	}
+	return conns, nil
+}
+
+// ParseConnections return the Connections from payload
+func ParseConnections(payload api.Payload) (conns []*Connections, err error) {
```

Comment:
<!-- Suggestions propose improvements to the current subject. It’s important to be explicit and clear on what is being suggested and why it is an improvement. -->
**💬 suggestion**
If you get a payload from the server via curl, you can save it to file and use it to unit test this function
Example: https://github.com/DataDog/datadog-agent/blob/main/test/fakeintake/aggregator/checkRunAggregator_test.go

Reply:
done Thanks